src/controllers/auth.js

```diff
@@ -71,3 +83,252 @@ export const register = async (req, res) => {
+  );
+
+  return formatResponse(res, StatusCodes.OK, {
+    message: "Password reset email sent",
+    info,
+  });
+};
+
+export const resetPassword = async (req, res) => {
+  const { id, token } = req.params;
+  const { password } = req.body;
+
+  const user = await findUserById(id);
+
+  if (!user)
+    return formatResponse(res, StatusCodes.NOT_FOUND, null, "User not found");
+
+  const isTokenValid = isTokenExpired(token);
+
+  if (isTokenValid)
+    return formatResponse(res, StatusCodes.UNAUTHORIZED, null, "Token expired");
+
+  const hashedPassword = await hashPassword(password);
+
+  const updatedUser = await updateUser(id, {
+    password: hashedPassword,
+  });
+
+  if (!updatedUser)
+    return formatResponse(
+      res,
+      StatusCodes.BAD_REQUEST,
+      null,
+      "Error updating user"
+    );
+
+  return formatResponse(res, StatusCodes.OK, {
+    message: "Password updated successfully",
+  });
+};
+
+export const getUsers = async (req, res) => {
+  const { companyId } = req.user;
+  try {
+    const users = await findAllUsers({
+      where: {
+        companyId: companyId,
+      },
+      attributes: [
+        "id",
+        "name",
+        "email",
+        "roleId",
+        "companyId",
+        "departmentId",
+        "isActive",
+        "profilePicture",
+        "createdAt",
+        "updatedAt",
+      ],
+    });
+
+    if (users.length === 0) {
+      return formatResponse(res, StatusCodes.NOT_FOUND, [], "Users not found");
+    }
+    return formatResponse(res, StatusCodes.OK, users);
+  } catch (error) {
+    return formatResponse(
+      res,
+      StatusCodes.INTERNAL_SERVER_ERROR,
+      null,
+      error.message
+    );
+  }
+};
+
+export const getUser = async (req, res) => {
+  const { id, companyId } = req.params;
+
+  try {
+    const user = await findUserById(
+      id,
+      {
+        companyId,
+      },
+      {
+        attributes: [
+          "id",
+          "name",
+          "email",
+          "roleId",
+          "companyId",
+          "departmentId",
+          "isActive",
+          "profilePicture",
+          "createdAt",
+          "updatedAt",
+        ],
+      }
+    );
+
+    if (!user) {
+      return formatResponse(res, StatusCodes.NOT_FOUND, [], "User not found");
+    }
+    return formatResponse(res, StatusCodes.OK, user);
+  } catch (error) {
+    return formatResponse(
+      res,
+      StatusCodes.INTERNAL_SERVER_ERROR,
+      null,
+      error.message
+    );
+  }
+};
+
+export const updateUserData = async (req, res) => {
+  const { id, companyId } = req.params;
+  const { name, email, password, profilePicture } = req.body;
+
+  try {
+    const user = await findUserById(id, {
+      companyId,
+    });
+
+    if (!user) {
+      return formatResponse(res, StatusCodes.NOT_FOUND, [], "User not found");
+    }
+
+    const hashedPassword = await hashPassword(password);
```

Comment:
Do you mean while updating the user the password always need to be provided?

Reply:
I changed the way to do so.